framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3094,15 +3095,35 @@ protected void checkThrownExpression(ThrowTree tree) {
         AnnotatedTypeMirror throwType = atypeFactory.getAnnotatedType(tree.getExpression());
         TypeMirror throwTM = throwType.getUnderlyingType();
         Set<? extends AnnotationMirror> required = getThrowUpperBoundAnnotations();
+        if (methodTree != null && getExceptionList(methodTree) != null) {
+            List<AnnotatedTypeMirror> exceptionList = getExceptionList(methodTree);
+            for (AnnotatedTypeMirror exception : exceptionList) {
+                Types typesUtil = atypeFactory.getProcessingEnv().getTypeUtils();
+                if (typesUtil.isSubtype(
```

Comment:
Can you add a comment why this only compares the underlying types?

Reply:
> Can you add a comment why this only compares the underlying types?

ok, I will add a comment here.